concent_api/conductor/tasks.py

```diff
@@ -0,0 +1,78 @@
+from celery import shared_task
+from mypy.types         import List
+
+from golem_messages     import message
+
+from .models            import VerificationRequest
+from .models            import UploadReport
+from .models            import UploadRequest
+
+
+@shared_task
+def verification_order_task(
+    subtask_id:             str,  # pylint: disable=unused-argument
+    src_code:               str,  # pylint: disable=unused-argument
+    extra_data:             str,  # pylint: disable=unused-argument
+    short_description:      str,  # pylint: disable=unused-argument
+    working_directory:      str,  # pylint: disable=unused-argument
+    performance:            str,  # pylint: disable=unused-argument
+    docker_images:          str,  # pylint: disable=unused-argument
+    source_file:            str,  # pylint: disable=unused-argument
+    result_file:            str,  # pylint: disable=unused-argument
+):
+    pass
+
+
+@shared_task
+def verification_request_task(
+    compute_task_def:   message.ComputeTaskDef,
+    files:              List[str],
+):
+    assert isinstance(compute_task_def, message.ComputeTaskDef)
+
+    # The app creates a new instance of VerificationRequest.
+    verification_request = VerificationRequest(
+        subtask_id          = compute_task_def['subtask_id'],
+        src_code            = compute_task_def['src_code'],
+        extra_data          = compute_task_def['extra_data'],
+        short_description   = compute_task_def['short_description'],
+        working_directory   = compute_task_def['working_directory'],
+        performance         = compute_task_def['performance'],
+        docker_images       = compute_task_def['docker_images'],
+    )
+    verification_request.full_clean()
+    verification_request.save()
+
+    # The app creates ne UploadRequest for each file listed in the task.
+    for file in files:
+        upload_request = UploadRequest(
+            path                 = file,
+            verification_request = verification_request,
+        )
+        upload_request.full_clean()
+        upload_request.save()
+
+        # If there are already UploadReports corresponding to some files, the app links them together by setting the
+        # value of the foreign key in UploadRequest.
+        UploadReport.objects.filter(
+            path            = file,
+            upload_request  = None,
+        ).update(
+            upload_request = upload_request
+        )
+
+    verification_request.refresh_from_db()
+
+    # If all expected files have been uploaded, the app sends verification_order task to the work queue.
+    if all([upload_request.upload_reports.exists() for upload_request in verification_request.upload_requests.all()]):
```

Comment:
Same as before, it is used only once and too small to make a method/function out of it.

Reply:
I suggested to make a method/function out of it not because of its size, but for better readability. I needed few seconds to figure out what this condition really means. A well named method/function would speed things up ;)

Reply:
I think that creating a helper variable is a good compromise in a situation like this. It gives us a name without the coding overhead of adding a new function for something that's not part of the API and used only once:

``` python
all_requests_have_reports = all(upload_request.upload_reports.exists() for upload_request in verification_request.upload_requests.all())

if all_requests_have_reports:
    # send verification task
```

BTW: You don't need to put the comprehension in `[]`. In fact it's better not to because `[]` creates an intermediate list holding all elements while without `[]` you get a generator expression. Not a huge difference when the list is small but it also reduces the amount of visual noise in the expression.